File: shared/database/models/notification.go
package models

type NotificationTag string

const (
	PostCreated     NotificationTag = "created"     // if post was created
	ObjectFound     NotificationTag = "found"       // if current user added found to a lost object and this last has been marked as found
	ObjectDelivered NotificationTag = "delivered"   // if claimed object has been delivered
	ClaimAdded      NotificationTag = "claim_added" // if lost object has been claimed
	FoundAdded      NotificationTag = "found_added" // if object has been claimed to be found by certain user
)

type Notification struct {
	ID     string          `gorm:"primaryKey" json:"id"`
	PostID string          `json:"course_id"`
	UserID string          `json:"user_id"`
	Post   *Post           `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE" json:"post,omitempty"`
	User   *User           `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE" json:"user,omitempty"`
	Seen   bool            `json:"seen"`
	Tag    NotificationTag `sql:"type:enum('created', 'found', 'delivered', 'claim_added', 'found_added')" gorm:"default:'created'" json:"tag"`
}

File: lib/shared/models/notification.dart
import 'package:estin_losts/shared/models/post.dart';

import 'user.dart';

enum NotificationTag { created, found, delivered }

class Notification {
  final String id;
  final String postID;
  final String userID;
  final Post post;
  final User user;
  final bool seen;
  final NotificationTag tag;

  Notification({
    required this.id,
    required this.postID,
    required this.userID,
    required this.post,
    required this.user,
    required this.seen,
    required this.tag,
  });

  factory Notification.fromJson(Map<String, dynamic> json) {
    return Notification(
      id: json['id'],
      postID: json['post_id'],
      userID: json['user_id'],
      post: json['post'],
      user: json['user'],
      seen: json['seen'],
      tag: json['tag'],
    );
  } 
}
